hyperspy/misc/date_time_tools.py

```diff
@@ -58,77 +50,125 @@ def get_date_time_from_metadata(metadata, formatting='ISO'):
+        dt = parser.parse('%s' % time).time()
+    elif date and not time:
+        dt = parser.parse('%s' % date).date()
+    else:
+        return
+
     if formatting == 'ISO':
         res = dt.isoformat()
     if formatting == 'datetime':
         res = dt
     # numpy.datetime64 doesn't support time zone
     if formatting == 'datetime64':
         res = np.datetime64('%sT%s' % (date, time))
-        
+
     return res
 
-        
+
 def update_date_time_in_metadata(dt, metadata):
-    """
-    TODO: documentation
+    """ Update the date and time in a metadata tree.
+
+        Parameters
+        ----------
+            dt : date and time information: it can be a ISO 8601 string,
+                a datetime.datetime or a numpy.datetime64 object
+            metadata : metadata object to update
+
+        Return
+        ----------
+            metadata object
+
+        Example
+        -------
+        >>> s = hs.load("example1.msa")
+        >>> dt = '2016-12-12T12:12:12-05:00'
+        >>> s.metadata = update_date_time_in_metadata(dt, s.metadata)
+        >>> s.metadata
+            ├── General
+            │   ├── date = 2016-12-12
+            │   ├── original_filename = example1.msa
+            │   ├── time = 12:12:12
+            │   ├── time_zone = 'EST'
+            │   └── title = NIO EELS OK SHELL
     """
     time_zone = None
     if isinstance(dt, str):
-        split = datetime.split('T')
-        date = split[0]
-        time = split[1]
-        if len(split) == 2:
-            time_zone = datetime.split('T')[2]
+        sp = dt.split('T')
+        date = sp[0]
+        time = sp[1]
+        if '+' in sp[1]:
+            time = "%s" % sp[1].split('+')[0]
+            time_zone = "+%s" % sp[1].split('+')[1]
+        elif '-' in sp[1]:
+            time = "%s" % sp[1].split('-')[0]
+            time_zone = "-%s" % sp[1].split('-')[1]
     if isinstance(dt, datetime.datetime):
         date = dt.date().isoformat()
         time = dt.time().isoformat()
-        if dt.tzname() is not None:
+        if dt.tzname():
             time_zone = dt.tzname()
-            metadata.set_item('General.time_zone', time_zone)
-    
+
     metadata.set_item('General.date', date)
     metadata.set_item('General.time', time)
+    if time_zone:
+        metadata.set_item('General.time_zone', time_zone)
     if metadata.has_item('General.time_zone') and not time_zone:
-        pass
-#        TODO: remove time_zone when necessary
+        del metadata.General.time_zone
     return metadata
 
-        
+
 def serial_date_to_ISO_format(serial):
+    """
+    Convert serial_date to a tuple of string (date, time, time_zone) in ISO 
+    format. By default, the serial date is converted in local time zone.
+    """
     # Excel date&time format
     origin = datetime.datetime(1899, 12, 30, tzinfo=tz.tzutc())
-    secs = (serial % 1.0) * 86400.0
+    secs = round(serial % 1.0 * 86400)
     delta = datetime.timedelta(int(serial), secs, secs / 1E6)
```

Comment:
I couldn't find any documentation on serial time really... so I inferred from what I found on the web! Please go ahead to change it.